doc/ethapp.adoc

```diff
@@ -1089,6 +1070,91 @@ _Output data_
 None
 
 
+### PROVIDE NETWORK CONFIGURATION
+
+#### Description
+
+This command handles the dynamic networks configuration, allowing to access funds
+without needing to update the application for each new network.
+
+This configuration must be send before any access to a network and stays valid until a new config is sent.
+Up to *2* different configurations can be used.The targeted slot is configured using P2.
+
+The configuration is sent in TLV (Tag-Length-Value) mode, whereas the icon itself is send as raw bytes in dedicated chunk(s).
+The configuration doesn't include the icon itself, but only its hash (`sha256`), and it is signed.
+
+For the configuration, the following tags are supported:
+
+[width="80%"]
+|===============================================================
+| *Tag name*        | *Tag value* | *Length (byte)*   | *Value*
+| STRUCTURE_TYPE    | 0x01        | 0x01              | 0x08 (for TYPE_DYNAMIC_NETWORK)
+| STRUCTURE_VERSION | 0x02        | 0x01              | 0x01
+| BLOCKCHAIN_FAMILY | 0x51        | 0x01              | 0x01 (for Ethereum)
+| CHAIN_ID          | 0x23        | 0x08              | (`uint64_t` value)
+| NETWORK_NAME      | 0x52        | variable (max 31) | string without '\0'
+| NETWORK_TICKER    | 0x24        | variable (max 10) | string without '\0'
+| NETWORK_ICON_HASH | 0x53        | 0x20              | _sha256_ of the network icon, sent separately
+| SIGNATURE         | 0x15        | variable          | Signature of the TLV payload data
+|===============================================================
+
+> Note For the icon:
+>
+>  - It is not supported on Nano devices and the corresponding APDU will be ignored.
+>
+>  - Depending on its size, it may be split on several chunks. The maximum icon buffer size allowed is `1KB`.
+>
+>  - Its value is the hex string generated by the script `<SDK_PATH>/lib_nbgl/tools/icon2glyph.py`, with parameter `--hexbitmap`
+
+The signature is mandatory, and is computed on the full payload data, using `CX_CURVE_SECP256K1`.
+
+#### Coding
+
+_Command_
+
+[width="80%"]
+|==============================================================
+| *CLA*     | *INS*    | *P1*                       | *P2*                     | *LC*     | *Le*
+.3+| E0  .3+| 30       | 00 : Network configuration | <slot>                   | variable | variable
```

Comment:
The fact that the wallet has to explicitly handle which slot it wants to use seems way overkill.
It could simply be a ring-buffer like it is for ERC-20 tokens.

Reply:
Updated